src/components/MainPage.jsx

```diff
@@ -12,68 +12,45 @@ import EventDetails from "./panels/EventDetails";
 import SpeakersPanel from "./panels/SpeakersPanel";
 import TalksPanel from "./panels/TalksPanel";
 import Navbar from "./Navbar";
+import { Routes, Route } from 'react-router-dom'
+import history from "../utilities/history";
 
 function MainPage() {
-  const [selectedTab, setSelectedTab] = useState("event");
+  // const [selectedTab, setSelectedTab] = useState("event");   // commented out because approach doesnt work with routing
 
-  const onSelectedTabChanged = (id) => {
-    setSelectedTab(id);
-  };
-
-  const showSelectedContent = () => {
-    switch (selectedTab) {
-      case "event":
-        return tabs[0].content;
-      case "speakers":
-        return tabs[1].content;
-      case "talks":
-        return tabs[2].content;
-      default:
-        return tabs[0].content;
-    }
-  };
+  // const onSelectedTabChanged = (id) => { // commented out because approache doesnt work with routing
+  //   setSelectedTab(id);
+  // };
 
   const tabs = [
     {
-      id: "event",
-      isSelected: selectedTab === "event",
+      id: "event",  
       label: "Event Details",
-      onClick: () => onSelectedTabChanged("event"),
-      content: (
-        <>
-          <EventDetails />
-        </>
-      ),
+      onClick: () => {history.push('/events'); window.location.reload();},
     },
     {
-      id: "speakers",
-      isSelected: selectedTab === "speakers",
+      id: "speakers",    
       label: "Speakers",
-      onClick: () => onSelectedTabChanged("speakers"),
-      content: (
-        <>
-          <SpeakersPanel />
-        </>
-      ),
+      onClick: () =>{history.push('/speakers'); window.location.reload();},
     },
     {
-      id: "talks",
-      isSelected: selectedTab === "talks",
+      id: "talks",   
       label: "Talks",
-      onClick: () => onSelectedTabChanged("talks"),
-      content: <TalksPanel />,
+      onClick: () => {history.push('/talks'); window.location.reload();},  
     },
   ];
 
-  const onLogoClick = useCallback(() => {
-    onSelectedTabChanged("event");
-  }, []);
+  
+  // const onLogoClick = useCallback(() => {  // commented out because it doesnt seem to work and conflicts with with routing 
```

Comment:
You can remove if this wasn't needed  - it probably stopped working when you commented out the `onSelectedTabChanged` which is how we were routing content before.